addons/calendar/models/calendar_attendee.py

```diff
@@ -24,54 +28,72 @@ def _default_access_token(self):
         ('accepted', 'Accepted'),
     ]
 
+    event_id = fields.Many2one('calendar.event', 'Meeting linked', required=True, ondelete='cascade')
+    partner_id = fields.Many2one('res.partner', 'Contact', required=True, readonly=True)
     state = fields.Selection(STATE_SELECTION, string='Status', readonly=True, default='needsAction',
                              help="Status of the attendee's participation")
     common_name = fields.Char('Common name', compute='_compute_common_name', store=True)
-    partner_id = fields.Many2one('res.partner', 'Contact', readonly=True)
-    email = fields.Char('Email', help="Email of Invited Person")
+    email = fields.Char('Email', related='partner_id.email', help="Email of Invited Person")
     availability = fields.Selection([('free', 'Free'), ('busy', 'Busy')], 'Free/Busy', readonly=True)
     access_token = fields.Char('Invitation Token', default=_default_access_token)
-    event_id = fields.Many2one('calendar.event', 'Meeting linked', ondelete='cascade')
+    recurrence_id = fields.Many2one('calendar.recurrence.rule', related='event_id.recurrence_id')
 
     @api.depends('partner_id', 'partner_id.name', 'email')
     def _compute_common_name(self):
         for attendee in self:
             attendee.common_name = attendee.partner_id.name or attendee.email
 
-    @api.onchange('partner_id')
-    def _onchange_partner_id(self):
-        """ Make entry on email and availability on change of partner_id field. """
-        self.email = self.partner_id.email
-
     @api.model_create_multi
     def create(self, vals_list):
         for values in vals_list:
+            if values.get('partner_id') == self.env.user.partner_id.id:
+                values['state'] = 'accepted'
             if not values.get("email") and values.get("common_name"):
                 common_nameval = values.get("common_name").split(':')
                 email = [x for x in common_nameval if '@' in x] # TODO JEM : should be refactored
                 values['email'] = email and email[0] or ''
                 values['common_name'] = values.get("common_name")
-        return super(Attendee, self).create(vals_list)
+        attendees = super().create(vals_list)
+        attendees._subscribe_partner()
+        return attendees
+
+    def unlink(self):
+        self._unsubscribe_partner()
+        return super().unlink()
+
+    def _subscribe_partner(self):
+        for event in self.event_id:
+            partners = (event.attendee_ids & self).partner_id - event.message_partner_ids
+            partners -= self.env.user.partner_id  # current user is automatically added as followers, don't add it twice.
```

Comment:
I think that it's managed by the ``message_subscribe`` method, but it doesn't harm anybody. :)